src/utils/project.ts

```diff
@@ -0,0 +1,78 @@
+import * as CryptoJS from 'crypto-js'
+
+import { createClient } from '@supabase/supabase-js'
+import { getConfig } from '../utils/config'
+
+const { supabaseUrl, supabaseApiKey, supabaseEncryptionKey } = getConfig()
+
+interface JwtSecretAndServiceApiKeys {
+  anonKey: string
+  jwtSecret: string
+  serviceKey: string
+}
+
+const jwtSecretAndServiceApiKeysCache: {
+  [projectRef: string]: JwtSecretAndServiceApiKeys
+} = {}
+
+async function getJwtSecretAndServiceApiKeys(
+  projectRef: string
+): Promise<JwtSecretAndServiceApiKeys> {
+  if (jwtSecretAndServiceApiKeysCache[projectRef]) {
+    return jwtSecretAndServiceApiKeysCache[projectRef]
+  }
+  if (!supabaseUrl) {
+    throw new Error('SUPABASE_URL environment variable is not set')
+  }
+  if (!supabaseApiKey) {
+    throw new Error('SUPABASE_API_KEY environment variable is not set')
+  }
+  const supabase = createClient(supabaseUrl, supabaseApiKey)
+  const {
+    data: {
+      jwt_secret_encrypted,
+      services: [{ service_api_keys }],
+    },
+  } = await supabase
+    .from('projects')
```

Comment:
hmm this couples the storage-api code to our internal middleware schema, lets see if we can come up with a better abstraction for this later on for those who want to run self hosted storage in multitenant mode

Reply:
As discussed offline, for a better abstraction, I'll add endpoints to add and delete per-tenant config.

Reply:
Done, please take a look.